### ISSUE TITLE
Brightfield and Darkfield create the opposite of what it should be?

### ISSUE THREAD

**Issue by weidtn** [DeepTrackAI/DeepTrack2#445]
I am simulating a MieSphere (Trying to simulate a Dynabead M270) in Brightfield and Darkfield. Normally, Brightfield microscopy should show a black particle in front of a white background; Darkfield should show a bright particle in front of a black background. This is also the behaviour we see at our microscope in the lab.

But in the simulation it is the opposite, just as it is in your example notebooks ([https://github.com/DeepTrackAI/DeepTrack2/blob/develop/tutorials/1-getting-started/DTGS121_tracking_particle_cnn.ipynb](https://github.com/DeepTrackAI/DeepTrack2/blob/develop/tutorials/1-getting-started/DTGS121_tracking_particle_cnn.ipynb)). So I guess this is intended? Why?

**Comment by giovannivolpe** [comment]
Whether the particle is bright or dark in brightfieqld depends on the parameters, see for example https://github.com/DeepTrackAI/DeepLearningCrashCourse/blob/main/Ch03_CNN/ec03_B_particle_localization/particle_localization.ipynb

Also, note that when using a single wavelength, our brightfieqld code really produces a holographic image of the particle.

If you can share your code, we can look into it in detail.

In any case, we will double-check the simulation in the next weeks.

**Comment by weidtn** [comment]
```python
import numpy as np
import deeptrack as dt
import matplotlib.pyplot as plt
import tqdm
import pyqtgraph as pg
import torch

if torch.cuda.is_available(): # This does not matter as the simulations are numpy only
    device = torch.device("cuda")
    print(f"Using GPU: {torch.cuda.get_device_name(0)}")
else:
    device = torch.device("cpu")
    print("Using CPU")

# Define Image
WIDTH = 200
HEIGHT = 200
NA=0.7
MAG = 50

z_px = np.sin(np.linspace(0, 4 * np.pi, 50)) * 50  # simulate z motion
y_px = np.full_like(z_px, HEIGHT // 2)
x_px = np.full_like(z_px, WIDTH // 2)
positions = np.stack([y_px, x_px, z_px], axis=-1)

r=1.4e-6
dT = 1e-3

### Rewrite this by using a deeptrack sequence for the positions:
### Define a function to draw a particle at a given position
def draw_particle(position, radius, width=1920, height=1080, magnification=50, n_wls=3):
    if n_wls < 1:
        raise ValueError("n_wls must be at least 1")
    elif n_wls == 1:
        wls = np.array([550e-9]) # Single wavelength at 550 nm
    wls = np.linspace(300, 800, n_wls) * 1e-9 # Take wavelengths from visible spectrum
    particle = dt.MieSphere(
        radius=radius,
        refractive_index=1.44,
        position=position,
        z=position[-1], # Why does it not work with the 3rd position?
        position_unit="pixel", # meter in z does not produce the same results as experiment
        intensity=1000,
    )
    # if torch.cuda.is_available():
    #     device = torch.device("cuda")
    #     particle = particle.torch(device=device)

    imgs = []
    for wl in wls:
        microscope = dt.Brightfield(
            NA=NA,
            wavelength=wl,
            magnification=MAG,
            output_region=[0, 0, HEIGHT, WIDTH],
            resolution=10e-6, # is this pixel size on sensor or in image?
            # z resolution should be ~1500nm, but this gives completely wrong images
            refractive_index_medium=1.33, 
        )
        img = microscope(particle) >> dt.NormalizeMinMax(0, 255)
        imgs.append(img.resolve())
    img_mean = sum(imgs) / len(imgs)
    return img_mean


transport = []
for pos in tqdm.tqdm(positions[::], desc="Generating images"):
    print(pos)
    img = draw_particle(
        position=pos, radius=r, width=WIDTH, height=HEIGHT, magnification=MAG, n_wls=5
    )
    transport.append(img)

# make an array with shape (time, x, y)
transport = np.array(transport).squeeze().transpose(0, 2, 1)

# Show video in pyqtgraph
img = pg.image(transport, title="Particle Transport", autoRange=True, autoLevels=True)

pg.exec()
```

This is what I am doing. I have correct trajectory data, but for now, I use constant x and y and oscillate z.
I use multiple wavelengths and expect a white background for Brightfield, but I get a black background. And Darkfield is showing artifacts with MieSphere, and most of the time, a bright background with a Sphere.

**Comment by Pwhsky** [comment]
Thanks for sharing your code, we are looking into the issue.
A few preliminary points that may already help clarify what you are observing:
1. **Darkfield should not be used with a regular `Sphere` object**
 Only MieSphere provides a physically correct scattering model for darkfield illumination.
 Using a simple geometric `Sphere` will produce incorrect background and artifacts. 
 We will add safeguards in a future release to prevent this unintended usage.

2. **The current darkfield implementation uses polarized illumination (x-polarized)**
 This is why the appearance may differ from unpolarized experimental setups.
 If you want unpolarized illumination, you need to simulate multiple polarization states and combine them.
 We are working on adding built-in support for unpolarized darkfield in an upcoming update.

3. **Normalization should be applied only after wavelength averaging**
 In your script, normalization is applied inside the wavelength loop. This distorts intensities and might reverse contrasts.

In addition regarding your comment about the `resolution` parameter when instancing the microscope:
`resolution` refers to the **camera pixel size**, if you want to use the _effective_ pixel size directly then you set `magnification = 1`.


We are reviewing the implementation in more detail.
